### PR TITLE
[autohint] updates to prepare for psautohint

### PR DESCRIPTION
**Pull request by josh-hadley**
- numerous updates to autohint.py and related modules (ufotools, beztools) to prepare for eventual replacement with psautohint. This is somewhat a work in progress towards completion of #826 
- updates to tests & test data for above
- some fixes in affected modules to fix errors reported by LGTM/Semmle

### CODE REVIEW

File: python/afdko/beztools.py [hunk 1]
```diff
@@ -1207,12 +1207,14 @@ def saveChanges(self):
 					tf.write(data)
 					tf.close()
 
-			elif  fontType == 2: # PS.
+			elif  2 <= fontType <= 3: # PS (PFA or PFB)
```
Comment (miguelsousa) [inline python/afdko/beztools.py]: would this be better?
```python
elif fontType in (2, 3):
```

File: tests/autohint_test.py [hunk 2]
```diff
@@ -81,7 +80,7 @@ def test_basic_hinting(font_filename, opt):
         actual_path = get_temp_file_path()
 
     diff_mode = []
-    if tail == '.cff':
+    if tail in {'.cff', '.pfb'}:
```
Comment (miguelsousa) [inline tests/autohint_test.py]: is there a particular reason to use a dict instead of a tuple?

Comment (josh-hadley) [reply tests/autohint_test.py]: That's a set, not dict:
```
>>> s = {1, 2}
>>> type(s)
<class 'set'>
```

File: python/afdko/beztools.py (hunk 1 above)
Comment (josh-hadley) [reply python/afdko/beztools.py]: Yes, that expresses the intent better, good suggestion. I updated in https://github.com/adobe-type-tools/afdko/pull/1011/commits/37a733c0a18267395d411761567573a820f4886a

**Pull request merged**